packages/terra-worklist-data-grid/src/WorklistDataGrid.jsx

```diff
@@ -62,158 +96,263 @@ function WorklistDataGrid(props) {
-    grid.current = node;
-    const focusedCell = grid.current.rows[focusedRow.current].cells[focusedCol.current];
-    focusedCell.tabIndex = 0;
-  }, []);
-
-  const handleKeyDown = (event) => {
-    const currentFocusedRow = event.target.parentElement.rowIndex;
-    const currentFocusedCol = event.target.cellIndex;
-
-    let nextRow = currentFocusedRow;
-    let nextCol = currentFocusedCol;
-
-    const key = event.keyCode;
-    switch (key) {
-      case KeyCode.KEY_UP:
-        nextRow -= 1;
-        break;
-      case KeyCode.KEY_DOWN:
-        nextRow += 1;
-        break;
-      case KeyCode.KEY_LEFT:
-        nextCol -= 1;
-        break;
-      case KeyCode.KEY_RIGHT:
-        nextCol += 1;
-        break;
-      case KeyCode.KEY_HOME:
-        nextCol = 0;
-        if (event.ctrlKey) {
-          nextRow = 1; // Assumption is that the first row is the column Heading.
-        }
-        break;
-      case KeyCode.KEY_END:
-        nextCol = props.rows[0].cells.length - 1;
-        if (event.ctrlKey) {
-          nextRow = props.rows.length;
-        }
-        break;
-      default:
-        return;
+  const [currentSelectedCell, setCurrentSelectedCell] = useState(null);
+  const [isNavigationEnabled, setIsNavigationEnabled] = useState(true);
+  const displayedColumns = (hasSelectableRows ? [WorklistDataGridUtils.ROW_SELECTION_COLUMN] : []).concat(columns);
+
+  const theme = useContext(ThemeContext);
+
+  const isRowSelectionCell = (columnIndex) => (
+    hasSelectableRows && columnIndex < displayedColumns.length && displayedColumns[columnIndex].id === WorklistDataGridUtils.ROW_SELECTION_COLUMN.id
+  );
+
+  const setFocus = (rowIndex, colIndex, makeActive) => {
+    let focusedCell = grid.current.rows[rowIndex].cells[colIndex];
+    if (isRowSelectionCell(colIndex) && focusedCell.firstChild) {
+      focusedCell = focusedCell.firstChild;
     }
-    if (nextRow >= grid.current.rows.length || nextCol >= grid.current.rows[0].cells.length) {
-      event.preventDefault();
-      return;
+    focusedCell.tabIndex = 0;
+    if (makeActive && focusedCell.focus) {
+      focusedCell.focus();
     }
-    if (nextCol < 0 || nextRow < 0) {
-      event.preventDefault();
-      return;
+  };
+
+  const removeTabStop = (rowIndex, colIndex) => {
+    const cell = grid.current.rows[rowIndex].cells[colIndex];
+    // Remove Tab stop from previous cell in table that has focus and set it to the
+    // cell that was clicked.
+    cell.tabIndex = -1;
+    if (isRowSelectionCell(colIndex) && cell.firstChild) {
+      // Also remove tabstop from the checkbox in the case of a row selection cell.
+      cell.firstChild.tabIndex = -1;
     }
+  };
+
+  const setFocusedRowCol = (newRowIndex, newColIndex, makeActive) => {
+    removeTabStop(focusedRow.current, focusedCol.current);
+    focusedRow.current = newRowIndex;
+    focusedCol.current = newColIndex;
+    setFocus(newRowIndex, newColIndex, makeActive);
+  };
```

Comment:
Do we really need to worry about the complexity of dealing with tab stops and focus with this implementation?  We really just need the active cell location stored in state and then use a class name to render the border.  It really isn't that the cell has focus, it is that it is the "active" cell.  Also, the current logic will break once you have descendants that can receive focus.  At that point your cell will no longer have focus.  You would need a special pseudo selector.